### DISCUSSION TITLE
Performance

### DISCUSSION THREAD

**Discussion by rserafim** [chakra-ui/chakra-ui#3140] [General] [answered]
Are there any performance differences in PageSpeed ​​if I install the complete Chakra or install only the components that I will use?

**Comment by rserafim** [comment]
Just adding ChakraProvider already loses 15 points in PageSpeed. Sad

**Reply by TomFreudenberg** [reply]
Same experience on my side - _+4 secs page load_

**Comment by Bertg** [comment, marked as answer]
I think this is more of a responsibility of the build logic. Webpack and other tools already provide "tree shaking" options which should remove all code except the code in use.

So this is really a "depends on your setup" kind of question, not really a Chakra one.

**Reply by rserafim** [reply]
> I think this is more of a responsibility of the build logic. Webpack and other tools already provide "tree shaking" options which should remove all code except the code in use.
> 
> So this is really a "depends on your setup" kind of question, not really a Chakra one.


Using standard NextJS + the Chakra.

**Reply by Bertg** [reply]
So, introducing a library like chakra into any exiting setup (even a bare bones one) is going to have an effect on the effective page speed. Your setup can help minimise this impact, but it will never bee as performant as hand written CSS & HTML.

Using Chakra - or any framework for that matter - will bring in some overhead. You trade of some initial performance loss for the richness of components and development speed.

If this is worth it to you, you must decide.

**Reply by rserafim** [reply]
Your explanation makes sense! Thanks
